From 749c3c52915aacc7137c80aad9cbb17700cc8678 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Thu, 30 Aug 2012 12:08:53 +0000
Subject: [PATCH 12/16] Fix
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=7679:
 Do not try to set IR header length when the profile
 is unknown

svn path=/trunk/; revision=44700
---
 epan/dissectors/packet-rohc.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/epan/dissectors/packet-rohc.c b/epan/dissectors/packet-rohc.c
index 0267b55..03a2bae 100644
--- a/epan/dissectors/packet-rohc.c
+++ b/epan/dissectors/packet-rohc.c
@@ -1414,8 +1414,10 @@ dissect_rohc_ir_packet(tvbuff_t *tvb, proto_tree *tree, packet_info *pinfo,
             break;
     }
 
-    /* Set length of IR header */
-    proto_item_set_len(ir_item, offset-ir_item_start);
+    if (offset != -1) {
+        /* Set length of IR header */
+        proto_item_set_len(ir_item, offset-ir_item_start);
+    }
 
     return offset;
 }
@@ -1737,7 +1739,7 @@ start_over:
         col_append_str(pinfo->cinfo, COL_INFO, "IR packet");
         offset = dissect_rohc_ir_packet(tvb, rohc_tree, pinfo, offset, cid, is_add_cid, p_rohc_info);
         if(offset == -1){
-            /* Could not pare header */
+            /* Could not parse header */
             return;
         }
         /*proto_tree_add_text(rohc_tree, tvb, offset, -1, "Data");*/
-- 
1.7.10.4

